docs/rewriter/rewrite_patterns.md

```diff
@@ -0,0 +1,74 @@
+# Pattern-based Rewrite Using Rules
+
+## Introduction
+
+The ONNX Rewriter tool provides the user with the functionality to replace certain patterns in an ONNX graph with another pattern based on rewrite rules provided by the user.
+
+## Usage
+
+There are three main components needed when rewriting patterns in the graph:
+
+1. `target_pattern` : Original pattern to match against. This pattern is written as a function using ONNXScript operators.
+2. `replacement_pattern` : Pattern to replace the original pattern with. This pattern is also written as a function using ONNXScript operators.
+3. `match_condition` (optional) : Pattern rewrite will occur only if the match condition is satisfied.
```

Comment:
At some place we should clarify our usage of the word `ONNXScript` ... what we mean is that we use the same style (from the users perspective), but internally the implementation is different, since it serves a different purpose.